lib/policies/index.js

```diff
@@ -10,12 +12,32 @@ class Policies {
       .readdirSync(path.resolve(__dirname))
       .filter(dir => fs.lstatSync(path.resolve(__dirname, dir)).isDirectory());
 
-    policyNames.forEach(policyName => {
-      this.policies[policyName] = require(path.resolve(__dirname, policyName));
+    policyNames.forEach((name) => {
+      const policy = require(path.resolve(__dirname, name));
+      this.policies[name] = Object.assign(policy, {
+        name
+      });
+      this.register(policy);
     });
   }
+
   register (policy) {
-    this.policies[policy.name] = policy;
+    const { name, schema } = policy;
+    const action = policy.policy;
+
+    const validate = schemas.register('policy', name, schema);
+
+    policy.policy = (params, ...args) => {
+      const validationResult = validate(params);
+      if (validationResult.isValid) {
+        return action(params, ...args);
+      } else {
+        logger.warn(`warning: policy ${name} params validation failed`, validationResult.error);
+        return null;
```

Comment:
Does this means that if I have incorrect schema I won't be able to use policy ? 
can it be  `return action(params, ...args);` here as well ?

Reply:
It depends — the idea behind the schema validation should be preventing the user from using the plugin if the provided parameters are not correct, therefore throwing an exception when this happens isn't it? Maybe @kevinswiber knows more about that?